db/jdbc/src/main/java/io/helidon/db/jdbc/JdbcStatement.java

```diff
@@ -0,0 +1,388 @@
+
+        try {
+            // Parameters names must be replaced with ? and names occurence order must be stored.
+            MappingParser parser = new MappingParser(statement);
+            String jdbcStatement = parser.convert();
+            LOGGER.finest(() -> String.format("Converted statement: %s", jdbcStatement));
+            PreparedStatement preparedStatement = connection.prepareStatement(jdbcStatement);
+            List<String> namesOrder = parser.namesOrder();
+            int i = 1;
+            for (String name : namesOrder) {
+                Object value = parameters.get(name);
+                LOGGER.info(String.format("Mapped parameter %d: %s -> %s", i, name, value));
+                preparedStatement.setObject(i, value);
+                i++;
+            }
+            return preparedStatement;
+        } catch (SQLException e) {
+            throw new DbException("Failed to prepare statement with named parameters: " + statementName, e);
+        }
+    }
+
+    private PreparedStatement prepareIndexedStatement(Connection connection,
+                                                      String statementName,
+                                                      String statement,
+                                                      List<Object> parameters) {
+
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(statement);
+            int i = 1; // JDBC set position parameter starts from 1.
+            for (Object value : parameters) {
+                LOGGER.info(String.format("Indexed parameter %d: %s", i, value));
+                preparedStatement.setObject(i, value);
+                // increase value for next iteration
+                i++;
+            }
+            return preparedStatement;
+        } catch (SQLException e) {
+            throw new DbException("Failed to prepare statement with indexed params: " + statementName, e);
+        }
+    }
+
+    /**
+     * Mapping parser state machine.
+     *
+     * Is it really worth it to write a parser, when we could use a regular expression with a single restriction,
+     * that the statement text must not contain :word except for parameters?
+     * The implementation is then really simple:
+     * <pre>
+     *  Pattern pattern = Pattern.compile(":(\\w+)", Pattern.UNICODE_CHARACTER_CLASS);
+     *  Matcher matcher = pattern.matcher(select);
+     *  List names = new LinkedList<>();
+     *  while(matcher.find()) {
+     *      names.add(matcher.group(1));
+     *  }
+     *  matcher.reset();
+     *  replaced = matcher.replaceAll("?");
+     *  </pre>
+     *
+     * Before replacement:
+     * {@code SELECT * FROM table WHERE name = :name AND type = :type}
+     * After replacement:
+     * {@code SELECT * FROM table WHERE name = ? AND type = ?}
+     * Expected list of parameteres:
+     * {@code "name", "type"}
+     */
+    private static final class MappingParser {
```

Comment:
Move to a separate class, create abstract ancestor in common. Rename methods.
Parser shall handle SQL comments too.
This is my own TODO to clean up the code.